app/UiTPAS/FetchNameForUdb3Organizer.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\UiTPAS;
+
+use App\Search\Sapi3\SearchService;
+use CultuurNet\SearchV3\ValueObjects\Organizer;
+
+/*
+ * If performance starts to be an issue, a cache would be good here.
+ * */
+
+final readonly class FetchNameForUdb3Organizer
```

Comment:
This feels like overkill compared to this https://github.com/cultuurnet/publiq-platform/blob/8ed64587e4d3db6ff2d92a57e3888b3dcc83b066/app/Domain/Integrations/GetIntegrationOrganizersWithTestOrganizer.php#L39

Reply:
If you want to make it reusable I would just make it part of `SearchService`

Reply:
I don't agree that it does 1 thing, it does 2 things. It gets the organizer and it gets the name. If you follow the SOLID principle, you would create a class that works on the Organizer collection and fetches the name of an already retrieved organizer.

Reply:
oké, I moved the dependency for the SearchService outside the class, and passed a collection.
Does this work for you?

Reply:
I would now also move this to the `Search` namespace and think about the name of the class and method. Something like `UdbOrganizerNameResolver::getName`